src/util/misc/projectStroke/StrokeLineJoinProjections.ts

```diff
@@ -215,22 +228,53 @@ export class StrokeLineJoinProjections extends StrokeProjectionsBase {
         Math.sqrt(newY ** 2 - ((newX * newY) / circleRadius.x) ** 2)
       );
 
-    [furthestX, furthestY].forEach((vector) => {
-      projections.push(
-        this.applySkew(this.A.add(vector)),
-        this.applySkew(this.A.subtract(vector))
-      );
-    });
+    [
+      furthestX,
+      furthestX.scalarMultiply(-1),
+      furthestY,
+      furthestY.scalarMultiply(-1),
+    ]
+      .map((vector) => this.applySkew(vector))
+      .forEach((vector) => {
+        if (isBetweenVectors(vector, startCircle, endCircle)) {
+          projections.push(this.applySkew(this.A).add(vector));
+        }
+      });
 
     return projections;
   }
 
   projectRound() {
+    const projections: Point[] = [];
+    /* Include the start and end points of the circle segment, so that only
+      the projections contained within it are included */
+    // add the orthogonal projections (start and end points of circle segment)
+    projections.push(...this.projectBevel());
+    // let's determines which one of the orthogonal projection is the beginning and end of the circle segment.
+    // when `alpha` equals 0 or 2*PI, we have a straight line, so the way to find the start/end is different.
+    const isStraightLine = this.alpha === 0 || this.alpha === twoMathPi,
+      // change the origin of the projections to point A
+      // so that the cross product calculation is correct
+      newOrigin = this.applySkew(this.A),
+      proj0 = projections[isStraightLine ? 0 : 2].subtract(newOrigin),
+      proj1 = projections[isStraightLine ? 1 : 0].subtract(newOrigin),
+      // when `isStraightLine` === true, we compare with the vector opposite AB, otherwise we compare with the bisector.
+      comparsionVector = isStraightLine
+        ? this.applySkew(this.AB.scalarMultiply(-1))
```

Comment:
Is this safe in case AB = 0?

Reply:
I think the only case that AB = 0 is when we have 1 point (which is handled in strokeLineCap). All others, reducedPoints will eliminate. am I wrong?

Reply:
Correct!